### PR TITLE
Feature/theming

### PR DESCRIPTION
**Pull request by ChawinTan**
## What's new

#272 

Added the custom theme colors provided in figma.

Theme colors as shown in storybook:
theme_colors

We can add on more custom theming as more specifications come in.

- Trying to make the e2e work in the meantime

## Self-checks

- [x] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [x] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Comment by matiasbavera** [comment]
Looks good on the reporting app as well
image

**Review by matiasbavera — COMMENTED** [review]
Code LGTM, well organized and it's working on the dashboard and on reporting. Do you want the fonts on this PR too (I saw that you ask this one =) )? 

Do you know why the e2e tests are failing?

**Comment by ChawinTan** [comment]
> Code LGTM, well organized and it's working on the dashboard and on reporting. Do you want the fonts on this PR too (I saw that you ask this one =) )?

Hmmmm, I think I will just add in the custom fonts that we are using now since they are only in 3 different places. Other than that, I don't think we have much to add since we are mostly following material ui's fonts. 

> Do you know why the e2e tests are failing?

There was a change from ros2-bridge to rmf-server, but this change wasn't updated in the rmf-models e2e yet. I tried cherry picking [this](https://github.com/open-rmf/rmf-web/commit/bf4b8085153d73ab5094504e34fbac0d1fbeb6f4) commit but there were too many conflicts. Since I am not too sure how `rmf-models` branch is going to develop and koon peng has said to leave it for now, I think we can leave it as it is.

**Comment by koonpeng** [comment]
rmf-models branch has been merged, can you rebase it to main?

**Comment by ChawinTan** [comment]
merged main into current branch

File: packages/react-components/lib/theme/theme.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    },
+  },
+  colors: {
+    black: '#000000',
+    xanadu: '#6B7D7D',
+    stPatricksBlue: '#103375',
+    blueYonder: '#5873A8',
+    ghostWhite: '#FBFCFF',
+    indianYellow: '#DEA54B',
+    fireOpal: '#F25F5C',
```
Comment (koonpeng) [inline packages/react-components/lib/theme/theme.tsx:50]: Are they any design specs for these colors? They are not part of the palette.

**Review by koonpeng — COMMENTED** [review]
Code looks ok, but I would like to wait for #302 and #307 to be merged first to see how it looks on the new layout.

File: packages/react-components/lib/theme/theme.tsx (hunk 1 above)
Comment (ChawinTan) [reply packages/react-components/lib/theme/theme.tsx:50]: I am following the color scheme provided in figma:
https://www.figma.com/file/nFiXHduImJev6xTodDpd0K/TNG-WIP?node-id=3%3A5814

**Comment by matiasbavera** [comment]
> Code looks ok, but I would like to wait for #302 and #307 to be merged first to see how it looks on the new layout.

Update:
blocked by:  https://github.com/open-rmf/rmf-web/pull/310 and #307

File: packages/react-components/lib/theme/theme.tsx (hunk 1 above)
Comment (koonpeng) [reply packages/react-components/lib/theme/theme.tsx:50]: These random colors without any design specs and not in the palette are very confusing. How do I know if/when I should use these colors vs the ones in the palette?

**Comment by ChawinTan** [comment]
Since I have already separated out the theme colors into main and dark, I will start working on a simple implementation for [this](https://github.com/open-rmf/rmf-web/issues/132) issue

**Pull request closed without merging**